### PR TITLE
Test ordering

### PR DESCRIPTION
**Pull request by hagberg**
Addresses some of #2203

### CODE REVIEW

**Comment by hagberg** [comment]
'Only' the algorithms/ modules are left to clean up the testing/look for bugs.  
I think this covers the rest.  
There are a couple fixes in here: sparse6.py, graph6.py, edgelist.py that should be reviewed.
Also the conversion from multidigraph to multigraph is problematical since it is traversal order dependent.

**Comment by jfinkels** [comment]
Might it be better to use `nose.tools.assert_items_equal`/`nose.tools.assert_count_equal` in order to avoid sorting everything?

File: networkx/generators/small.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def LCF_graph(n,shift_list,repeats,create_using=None):
     # start with the n-cycle
     G=cycle_graph(n, create_using)
     G.name="LCF_graph"
-    nodes = list(G)
+    nodes = sorted(list(G))
```
Comment (jfinkels) [inline networkx/generators/small.py]: This can just be `sorted(G)`.

**Comment by hagberg** [comment]
Yes, we can probably simply some of the tests to remove the `sorted` calls.

File: networkx/readwrite/edgelist.py [hunk 2]
```diff
@@ -109,8 +109,13 @@ def generate_edgelist(G, delimiter=' ', data=True):
     --------
     write_adjlist, read_adjlist
     """
-    if data is True or data is False:
-        for e in G.edges(data=data):
+    if data is True:
```
Comment (jfinkels) [inline networkx/readwrite/edgelist.py]: I don't quite understand this change. Can you explain what's going on here?

Comment (hagberg) [inline networkx/readwrite/edgelist.py]: Looks like i wanted to make a copy of the data dictionary.  Does make_str modify in place?

Comment (jfinkels) [inline networkx/readwrite/edgelist.py]: Looks like it just returns `str(x)` (or `unicode(str(x), 'unicode-escape')` on Python 2), so no.

**Comment by hagberg** [comment]
I'm going to merge these - the work is note complete and can continue (fixing tests in the algorithms/ package) in #2203.

**Pull request merged**